static/js/compose.js

```diff
@@ -627,6 +628,38 @@ export function initialize() {
         }
     });
 
+    let instance = null;
+    $("body").on("click", ".time_pick", (e) => {
+        e.preventDefault();
+
+        if (instance === null) {
+            const on_timestamp_selection = (val) => {
+                const timestr = `<time:${val}> `;
+                compose_ui.insert_syntax_and_focus(timestr);
```

Comment:
The message editing bug turns out to be that we need to pass the second parameter to this function; I went ahead and fixed that.